rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -1018,17 +1018,17 @@ namespace rsx
 					return;
 			}
 
-			std::unique_ptr<u8[]> temp1, temp2, temp3, sw_temp;
+			std::vector<u8> temp1, temp2, temp3, sw_temp;
 
 			if (scale_y < 0 || scale_x < 0)
 			{
-				temp1.reset(new u8[in_pitch * (in_h - 1) + (in_bpp * in_w)]);
+				temp1.resize(in_pitch * in_h);
```

Comment:
What if in_pitch < in_bpp * in_w? can't it result in requested size of allocation lower than needed?

Reply:
That is an impossible definition of a 2D region. I can add an assert for it though.

Reply:
Maybe its not useful for games but im pretty sure it is allowed.

Reply:
I tested it, its allowed.

Reply:
For clarification its only about in_pitch, with out_pitch its causing an exception to be raised.

Reply:
This is old code. I rewrote it in a better way. There is still another somewhat unrelated bug but I have commits for them locally.